### PR TITLE
Adds drop_duplicates

### PR DESCRIPTION
**Pull request by robertkossendey**
This PR adds a new function, drop_duplicates, that drops all but one duplicate row from a Delta table. 
This relates to Issue  #3 .

### CODE REVIEW

File: mack/__init__.py [hunk 1]
```diff
@@ -105,6 +108,47 @@ def kill_duplicates(deltaTable, pkey, cols):
...
 
 
+def drop_duplicates(delta_table: DeltaTable, primary_key: str, duplication_columns: List[str]):
+    if not delta_table:
+        raise Exception("An existing delta table must be specified.")
+
+    if not primary_key:
+        raise Exception("A primary key must be specified.")
+
+    if not duplication_columns or len(duplication_columns) == 0:
```
Comment (scottsand-db) [inline mack/__init__.py:118]: can the `primary_key` be the only column in `duplication_columns`? we should probably prevent that

Comment (robertkossendey) [reply mack/__init__.py:118]: Do you mean like `mack.drop_duplicates(deltaTable, "col1", ["col1"])`? I will check that duplication_columns and primary_key are mutually exclusive

Comment (scottsand-db) [reply mack/__init__.py:118]: Yup that's what I mean!

**Comment by robertkossendey** [comment]
@scottsand-db updated the PR accordingly.

**Pull request merged**